applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -416,6 +416,36 @@ bool CheckWakeCondition(const Element& rElement, const double& rTolerance, const
     return wake_condition_is_fulfilled;
 }
 
+template <int Dim, int NumNodes>
+double ComputeLocalMachSquaredDerivative(
+    const array_1d<double, Dim>& velocity,
+    const double& local_mach_number,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // Following Fully Simulataneous Coupling of the Full Potential Equation
+    //           and the Integral Boundary Layer Equations in Three Dimensions
+    //           by Brian Nishida (1996), Section A.2
+
+    // read free stream values
+    const double heat_capacity_ratio = rCurrentProcessInfo[HEAT_CAPACITY_RATIO];
+    const double free_stream_mach = rCurrentProcessInfo[FREE_STREAM_MACH];
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+
+    // make squares of values
+    double sq_local_mach_number = pow(local_mach_number, 2);
+    double sq_free_stream_mach = pow(free_stream_mach, 2);
+    double sq_local_velocity = inner_prod(velocity, velocity);
+    double sq_free_stream_velocity = inner_prod(free_stream_velocity, free_stream_velocity);
+
+    // square bracket term
+    double square_bracket_term = 1.0 + 0.5*(heat_capacity_ratio - 1)*
+            sq_free_stream_mach*(1 - (sq_local_velocity/sq_free_stream_velocity));
+
+    // calculate mach derivative
+    return sq_local_mach_number * ((1/sq_local_velocity) +
+            0.5*(heat_capacity_ratio - 1)*(1/sq_free_stream_velocity)*sq_free_stream_mach*(1/square_bracket_term));
```

Comment:
```suggestion
    double square_bracket_term = 1.0 + 0.5*(heat_capacity_ratio - 1.0)*
            sq_free_stream_mach*(1.0 - (sq_local_velocity/sq_free_stream_velocity));

    // calculate mach derivative
    return sq_local_mach_number * ((1.0/sq_local_velocity) +
            0.5*(heat_capacity_ratio - 1.0)*(1.0/sq_free_stream_velocity)*sq_free_stream_mach*(1.0/square_bracket_term));
```

be carefuls with ints and doubles, mixing them can lead to unpleasant surprises